src/envs/pybullet_cover.py

```diff
@@ -0,0 +1,313 @@
+        [np.pi / 2, np.pi / 2, -np.pi])
+    _move_to_pose_tol: ClassVar[float] = 1e-7
+
+    # Object parameters.
+    _obj_len_hgt: ClassVar[float] = 0.045
+    _max_obj_width: ClassVar[float] = 0.07  # highest width normalized to this
+
+    # Dimension and workspace parameters.
+    _table_height: ClassVar[float] = 0.2
+    _y_lb: ClassVar[float] = 0.4
+    _y_ub: ClassVar[float] = 1.1
+    _robot_init_y: ClassVar[float] = (_y_lb + _y_ub) / 2
+    _pickplace_z: ClassVar[float] = _table_height + _obj_len_hgt * 0.5 + 0.01
+    _target_height: ClassVar[float] = 0.0001
+
+    def __init__(self) -> None:
+        super().__init__()
+
+        # Override PickPlace option
+        types = self._PickPlace.types
+        params_space = self._PickPlace.params_space
+        toggle_fingers_func = lambda s, _1, _2: (
+            (self._open_fingers, self._closed_fingers)
+            if self._HandEmpty_holds(s, []) else
+            (self._closed_fingers, self._open_fingers))
+        self._PickPlace: ParameterizedOption = \
+            utils.LinearChainParameterizedOption(
+                "PickPlace",
+                [
+                    # Move to far above the location we will pick/place at.
+                    self._create_cover_move_option(
+                        name="MoveEndEffectorToPrePose",
+                        target_z=self._workspace_z),
+                    # Move down to pick/place.
+                    self._create_cover_move_option(
+                        name="MoveEndEffectorToPose",
+                        target_z=self._pickplace_z),
+                    # Toggle fingers.
+                    self._pybullet_robot.create_change_fingers_option(
+                        "ToggleFingers", types, params_space,
+                        toggle_fingers_func),
+                    # Move back up.
+                    self._create_cover_move_option(
+                        name="MoveEndEffectorBackUp",
+                        target_z=self._workspace_z)
+                ])
+        self._block_id_to_block: Dict[int, Object] = {}
+        self._target_id_to_target: Dict[int, Object] = {}
+
+    def _initialize_pybullet(self) -> None:
+        """Run super(), then handle cover-specific initialization."""
+        super()._initialize_pybullet()
+
+        # Load table in both the main client and the copy.
+        self._table_id = p.loadURDF(
+            utils.get_env_asset_path("urdf/table.urdf"),
+            useFixedBase=True,
+            physicsClientId=self._physics_client_id)
+        p.resetBasePositionAndOrientation(
+            self._table_id,
+            self._table_pose,
+            self._table_orientation,
+            physicsClientId=self._physics_client_id)
+        p.loadURDF(utils.get_env_asset_path("urdf/table.urdf"),
+                   useFixedBase=True,
+                   physicsClientId=self._physics_client_id2)
+        p.resetBasePositionAndOrientation(
+            self._table_id,
```

Comment:
don't we need to use the Id from the second client here? what guarantees that they are the same? (realizing this may be an issue in the PR we already merged too)